git-autofixup

```diff
@@ -119,10 +119,67 @@ sub git_cmd {
     return ('git', @GIT_OPTIONS, @_);
 }
 
+# With a linear git history there'll be a single merge base that's easy to
+# refer to with @{upstream}, but during an interactive rebase we need to get
+# the "current" branch from the rebase metadata.
+#
+# Unusual cases:
+#
+# While there can be multiple merge bases if there have been criss-cross
+# merges, there'll still be a single fork point unless the relevant reflog
+# entries have already been garbage-collected.
+#
+# When multiple upstreams are configured via `branch.<name>.merge` in git's
+# config the most correct approach is probably to find the fork-point for each
+# merge value and return those. But it seems unlikely that someone is doing
+# octopus merges and using git-autofixup, so we're not handling that specially
+# currently.
+sub find_merge_bases {
+    my $upstream = '@{upstream}';
+
+    # If an interactive rebase is in progress, derive the upstream from the
+    # rebase meatadata.
+    my $gitdir = qx(git rev-parse --git-dir) or die "git rev-parse: $!\n";
+    chomp $gitdir;
+    if (-e "$gitdir/rebase-merge") {
+         my $branch = slurp("$gitdir/rebase-merge/head-name");
+         chomp $branch;
+         $branch =~ s#^refs/heads/##;
+         $upstream = "$branch\@{upstream}";
+    }
+
+    # `git merge-base` will fail if there's no tracking branch. In that case
+    # redirect stderr and communicate failure by returning an empty list. Also,
+    # with the --fork-point option, no merge bases are returned if the relevant
+    # reflog entries have been GC'd, so fall back to normal merge-bases.
+    my @merge_bases;
+    for (qx(git merge-base --all --fork-point $upstream HEAD 2>/dev/null)) {
+        chomp;
+        push @merge_bases, $_;
+    }
+    if (!@merge_bases) {
+        for (qx(git merge-base --all $upstream HEAD 2>/dev/null)) {
```

Comment:
I guess here the ideal error output here would be something like

    git merge-base: fatal: no upstream configured for branch 'my-branch'
    Can't find tracking branch. Please specify a revision.

Reply:
Yeah, for commands that are expected to fail it's probably worth prefixing their stderr lines with the command (as you showed) so we can pass errors through while making their provenance obvious.